jerry-core/ecma/base/ecma-gc.c

```diff
@@ -231,16 +231,30 @@ ecma_gc_mark_promise_object (ecma_extended_object_t *ext_object_p) /**< extended
 
   while (ecma_value_p != NULL)
   {
-    ecma_gc_set_object_visited (ecma_get_object_from_value (*ecma_value_p));
-    ecma_value_p = ecma_collection_iterator_next (ecma_value_p);
+    if (ecma_is_value_object (*ecma_value_p))
+    {
+      ecma_gc_set_object_visited (ecma_get_object_from_value (*ecma_value_p));
+      ecma_value_p = ecma_collection_iterator_next (ecma_value_p);
+    }
+    else
+    {
+      break;
```

Comment:
My two cents here: isn't this an overly complicated way of writing
```c
while (ecma_value_p != NULL && ecma_is_value_object (*ecma_value_p))
```
? The `if` in the `while` is only justified if there is no `else` with a `break`, IMO. (I think that's what @zherczeg was asking.)